benchmark_runner/common/oc/oc.py

```diff
@@ -68,32 +70,39 @@ def get_pv_disk_ids(self):
         result = self.run(f"{self.__cli} get pv -o jsonpath={{.items[*].metadata.annotations.'storage\.openshift\.com/device-id'}}")
         return result.split()
 
+    def remove_lso_path(self):
+        """
+        The method removes lso path on each node
+        @return:
+        """
+        self.run(fr"""{self.__cli} get nodes -l node-role.kubernetes.io/worker= -o jsonpath="{{range .items[*]}}{{.metadata.name}}{{'\\n'}}{{end}}" |  xargs -I{{}} oc debug node/{{}} -- chroot /host sh -c "rm -rf /mnt/local-storage/local-sc" """)
+
     def get_worker_disk_ids(self):
         """
         The method returns worker disk ids
         """
         workers_disk_ids = []
         if self.__worker_disk_ids:
-            for node, disk_ids in self.__worker_disk_ids.items():
-                for disk_id in disk_ids:
-                    workers_disk_ids.append(disk_id)
+            workers_disk_ids = [disk_id for disk_list in self.__worker_disk_ids.values() for disk_id in disk_list]
         return workers_disk_ids
 
-    def remove_lso_path(self):
-        """
-        The method removes lso path on each node
-        @return:
-        """
-        self.run(fr"""{self.__cli} get nodes -l node-role.kubernetes.io/worker= -o jsonpath="{{range .items[*]}}{{.metadata.name}}{{'\\n'}}{{end}}" |  xargs -I{{}} oc debug node/{{}} -- chroot /host sh -c "rm -rf /mnt/local-storage/local-sc" """)
-
     def get_free_disk_id(self):
         """
-        This method returns free disk (workers_all_disk_ids - workers_odf_pv_disk_ids)
+        This method extracts the free disk by worker_disk_ids - pv_disk_ids
+        :return: It returns the first free disk ID with the 'scsi-3600062b' prefix
         """
-        workers_disk_ids = self.get_worker_disk_ids()
-        workers_pv_disk_ids = self.get_pv_disk_ids()
-        if workers_disk_ids:
-            return list(set(workers_disk_ids) - set(workers_pv_disk_ids))
+        free_disk_id = ''
+        if self.get_worker_disk_ids():
+            workers_disk_ids = [disk_id for disk_id in self.get_worker_disk_ids()]
+            workers_pv_disk_ids = [disk_id[self.DISK_ID_PREFIX_LEN:] for disk_id in self.get_pv_disk_ids()]
+            # Calculate the set difference
+            unique_disk_ids = list(set(workers_disk_ids) - set(workers_pv_disk_ids))
```

Comment:
Rather than this, you need something like

```
unique_disk_ids = []
for worker_disk in workers_disk_ids:
    disk_matched = False
    for pv_disk in workers_pv_disk_ids:
        if workers_disk.endsWith(pv_disk):
            disk_matched=True
            break
    if not disk_matched:
        unique_disk_ids.append(prefix + worker_disk
```

or some such.

Reply:
Note that when the prefix was `google-` it was 7 bytes, rather than 6.  You cannot make assumptions about the prefix!

Reply:
You don't need to.  All you need to do is write the code to be independent of type, by simply matching the unprefixed ID against the list of IDs by using `endsWith`